safenode/src/bin/kadnode.rs

```diff
@@ -1,233 +1,178 @@
-                    //         //         "Got record {:?} {:?}",
-                    //         //         std::str::from_utf8(key.as_ref()).unwrap(),
-                    //         //         std::str::from_utf8(&value).unwrap(),
-                    //         //     );
-                    //         // }
-                    //         // QueryResult::GetRecord(Ok(_)) => {}
-                    //         // QueryResult::GetRecord(Err(err)) => {
-                    //         //     eprintln!("Failed to get record: {err:?}");
-                    //         // }
-                    //         // QueryResult::PutRecord(Ok(PutRecordOk { key })) => {
-                    //         //     println!(
-                    //         //         "Successfully put record {:?}",
-                    //         //         std::str::from_utf8(key.as_ref()).unwrap()
-                    //         //     );
-                    //         // }
-                    //         // QueryResult::PutRecord(Err(err)) => {
-                    //         //     eprintln!("Failed to put record: {err:?}");
-                    //         // }
-                    //         // QueryResult::StartProviding(Ok(AddProviderOk { key })) => {
-                    //         //     println!(
-                    //         //         "Successfully put provider record {:?}",
-                    //         //         std::str::from_utf8(key.as_ref()).unwrap()
-                    //         //     );
-                    //         // }
-                    //         // QueryResult::StartProviding(Err(err)) => {
-                    //         //     eprintln!("Failed to put provider record: {err:?}");
-                    //         // }
-                    //         _ => {
-                    //             //
-                    //         }
-                    //     }
-                    // }
-                    _ => {}
                 }
-
             }
         }
     });
 
-    sender
-}
-
-#[tokio::main]
-async fn main() -> Result<()> {
-    let log_dir = grab_log_dir();
-    let _log_appender_guard = init_node_logging(&log_dir)?;
-
-    info!("start");
-    let channel = run_swarm();
-
-    let x = xor_name::XorName::from_content(b"some random content here for you");
-
-    if let Err(e) = channel.send(SwarmCmd::Search(x)).await {
-        debug!("Error while seding SwarmCmd: {e}");
+    // wait until we discover atleast one peer
+    peer_dicovered_rx.await?;
+    info!("Discovered a Peer");
+    // todo: sometimes, the node might query the network before it adds a peer to the DHT. The
+    // PeerDiscoverd event is triggered when it adds the peer to the DHT, but the op might fail and
+    // there is no way to confirm it since `RoutingUpdate` is private/no debug impl.
+    // Hence sleep for sometime before querying the network
+    thread::sleep(time::Duration::from_millis(100));
+
+    if let Some(files_path) = opt.upload_chunks {
```

Comment:
It was surprising to me that the variable `upload_chunks` containied a `files_path`.

Reply:
The `kadnode.rs` is just a prototype file for now, as we had discussed in slack :+1: